client/my-sites/stats/stats-post-detail/index.jsx

```diff
@@ -92,34 +92,69 @@ class StatsPostDetail extends Component {
 			return translate( 'Home page / Archives' );
 		}
 
-		if ( typeof post?.title === 'string' && post.title.length ) {
+		if ( typeof post?.title === 'string' && post?.title.length ) {
 			return decodeEntities( stripHTML( post.title ) );
 		}
 
-		if ( typeof postFallback?.post_title === 'string' && postFallback.post_title.length ) {
+		if ( typeof postFallback?.post_title === 'string' && postFallback?.post_title.length ) {
 			return decodeEntities( stripHTML( postFallback.post_title ) );
 		}
 
 		return null;
 	}
 
+	getPost() {
+		const { isPostHomepage, post, postFallback } = this.props;
+
+		const postBase = {
+			title: this.getTitle(),
+			type: isPostHomepage ? 'page' : 'post',
+		};
+
+		// Check if post is valid.
+		if ( typeof post === 'object' && post?.title.length ) {
+			return Object.assign( postBase, {
+				date: post?.date,
+				post_thumbnail: post?.post_thumbnail,
+				like_count: post?.like_count,
+				comment_count: post?.discussion?.comment_count,
+				type: post?.type,
+			} );
```

Comment:
Alternatively, we can use object spread:

```suggestion
			return {
				...postBase,
				date: post?.date,
				post_thumbnail: post?.post_thumbnail,
				like_count: post?.like_count,
				comment_count: post?.discussion?.comment_count,
				type: post?.type,
			} );
```

Note that invoking [`Object.assign( objectA, { ...objectB } )` will **mutate** `objectA`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).